imagedephi/redact.py

```diff
@@ -90,6 +161,96 @@ def execute_plan(self) -> None:
                 rule.apply(ifd)
 
 
+class SvsMetadataRedactionPlan(TiffMetadataRedactionPlan):
+    """
+    Represents a plan of action for redacting files in Aperio (.svs) format.
+
+    Redaction for this type of file is similar to redaction for .tiff files, as the
+    formats are similar. However, Aperio images store additional information in its
+    ImageDescription tags. As a result, this tag is treated specially here.
+    """
+
+    file_format = FileFormat.SVS
+    description_redaction_steps: dict[str, MetadataSvsRule]
+    no_match_description_keys: set[str]
+
+    def __init__(
+        self,
+        tiff_info: TiffInfo,
+        base_rules_tiff: list[MetadataTiffRule],
+        base_rules_svs: list[MetadataSvsRule],
+        override_rules_tiff: list[MetadataTiffRule],
+        override_rules_svs: list[MetadataSvsRule],
+    ) -> None:
+        super().__init__(tiff_info, base_rules_tiff, override_rules_tiff)
+
+        image_description_tag = tifftools.constants.Tag["ImageDescription"]
+        del self.redaction_steps[image_description_tag.value]
+
+        self.description_redaction_steps = {}
+        self.no_match_description_keys = set({})
+        ifds = self.tiff_info["ifds"]
+        for tag, ifd in self._iter_tiff_tag_entries(ifds):
+            if tag.value != image_description_tag.value:
+                continue
+
+            svs_description = SvsDescription(str(ifd["tags"][tag.value]["data"]))
+            for key in svs_description.metadata.keys():
+                for rule in chain(override_rules_svs, base_rules_svs):
+                    if rule.is_match(key):
+                        self.description_redaction_steps[key] = rule
+                        break
+                else:
+                    self.no_match_description_keys.add(key)
+
+    def is_comprehensive(self) -> bool:
+        return super().is_comprehensive() and len(self.no_match_description_keys) == 0
+
+    def report_missing_rules(self) -> None:
+        if self.is_comprehensive():
+            click.echo("The redaction plan is comprehensive.")
+        else:
+            if self.no_match_tags:
+                click.echo(
+                    "The following tags could not be redacted given the current set of rules."
+                )
+                for tag in self.no_match_tags:
+                    click.echo(f"{tag.value} - {tag.name}")
+            if self.no_match_description_keys:
+                click.echo(
+                    "The following keys were found in Aperio ImageDescription strings "
+                    "and could not be redacted given the current set of rules."
+                )
+                for key in self.no_match_description_keys:
+                    click.echo(key)
```

Comment:
It'd be nice to make this output more grepable -- when I run on a huge list of files it'd be nice to have "Missing tag, Aperio: {key}" to filter on. (and "Missing tag, tiff: {key}" in the tiff output).

Also, I don't know why we are using `click.echo` instead of `print`.  We aren't using *any* of the coloring or other features of click, so this just makes the code harder to read.